File: ruby/raindrops/raindrops.rb
class Raindrops
  RAIN_TRANSLATOR = {3=>'Pling',
                     5=>'Plang',
                     7=>'Plong'}
  def self.convert(number)
    rain_output = ''
    RAIN_TRANSLATOR.each do |key,value|
        rain_output += value if number % key == 0
    end
    rain_output.empty? ? number.to_s : rain_output
  end
end

module BookKeeping
  VERSION = 3
end

File: python/raindrops/raindrops.py
SOUNDS = {
    3: "Pling",
    5: "Plang",
    7: "Plong",
}

def convert(number):
    sound = "".join([find_sound(number, factor) for factor in SOUNDS.keys()])
    if sound == "":
        return str(number)

    return sound

def find_sound(number, factor):
    if ((number % factor) == 0):
        return SOUNDS.get(factor)

    return ""
